lib/bullet.js

```diff
@@ -15,47 +15,82 @@ Bullet.prototype.draw = function () {
   this.context.arc(this.x, this.y, this.radius, 2 * Math.PI, false);
   this.context.fillStyle = "#000000";
   this.context.fill();
+
   return this;
 };
 
-Bullet.prototype.move = function (obstacle) {
-  if(this.x > this.context.canvas.width - this.width){
-    this.directionX = -this.directionX;
-    this.bounceCount++;
-  }
-  if(this.x < 0){
-    this.directionX = -this.directionX;
-    this.bounceCount++;
-  }
-  if(obstacle.height > 10 && this.x > obstacle.x - obstacle.width && this.x < obstacle.x + obstacle.width * 2 && this.y >= obstacle.y && this.y < obstacle.y + obstacle.height && obstacle.y > 0){
-    this.directionX = -this.directionX;
-    this.bounceCount++;
-  }
-  if(obstacle.height > 10 && this.x > obstacle.x - obstacle.width && this.x < obstacle.x + obstacle.width * 2 && obstacle.y === 0 && this.y < obstacle.height){
-    this.directionX = -this.directionX;
-    this.bounceCount++;
-  }
+
+Bullet.prototype.move = function (obstacles) {
+  this.collisionCheck(obstacles);
   this.x += this.directionX;
+  this.y += this.directionY;
+
+  return this;
+};
+
 
-  if(this.y > this.context.canvas.height - this.height){
-    this.directionY = -this.directionY;
+Bullet.prototype.collisionCheck = function (obstacles) {
+  var posX = this.x + this.radius;
+  var posY = this.y + this.radius;
+
+  // check collision with left & right canvas borders
+  if(posX >= this.context.canvas.width || posX <= this.radius) {
+    this.directionX *= -1;
```

Comment:
^^ You'll see this a few times. I changed it from `this.directionX = -this.directionX` to `this.directionX *= -1` because I thought it looked a little cleaner since the line is slightly shorter. It's still doing the same thing as before, because multiplying it by -1 simply changes the value from positive to negative (or vice versa)